lib/at_ex/USSD.ex

```diff
@@ -0,0 +1,36 @@
+defmodule AtEx.USSD do
+  @moduledoc """
+  This module handles building valid response for the Africas Talking USSD endpoint
+  """
+
+  @doc """
+  This function accepts a list of string and proceeds to build a valid response for
+  the AT API
+  """
+  @spec build_response(list(String.t())) :: {:ok, String.t()} | {:error, term()}
```

Comment:
The data structure we expect from a user, seems can be more complicated than a list. Looking at the sample example code [AT ussd code sample](https://build.at-labs.io/docs/ussd%2Foverview) I feel a list with maps would suffice.  for instance like the sample code I would have this `data = [
  %{"CON What would you want to check" => 
    %{
      "My Account" => %{ "CON Choose account information you want to view" =>
        %{"Account Number" => %{:account_number => "END Your account number is ACC1001"}, "Account Balance" => %{ :account_balance => " END Your balance is 100000"}}} ,
      "My Phone Number" => %{:phone_number => "END your phone number is 07200000"}
    }
  }
]` as the data structure.

Reply:
So an app can send that to the API and we start generating a response based on that, as a requirement we will ask developers to give us data that is a list having one map value that can have many other inner maps depending on how deep their USSD logic is.

Reply:
@kamalogudah  I really think this is mixing concerns, remember we are building an API wrapper, above you are conflating business logic with response building, remember we don't build multiple responses, we build one response at a time also, please show how the above becomes a valid AT response

Reply:
@zacck mmmmh I see, I think I get the point, was not clear with that one response at a time, this makes things clear, thanks for the clarification.